File: blukutukhttp/src/main/java/com/robinyonathan/blukutukhttp/Network.java
package com.robinyonathan.blukutukhttp;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

class Network {
    static boolean isNetworkAvailable(Context context) {
        ConnectivityManager connectivityManager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);

        boolean connected = false;

        if (connectivityManager != null) {
            NetworkInfo activeNetworkInfo = connectivityManager.getActiveNetworkInfo();
            if (activeNetworkInfo != null) {
                connected = activeNetworkInfo.isConnected();
            }
        }

        return connected;
    }
}

File: blukutukhttp/src/main/java/com/robinyonathan/blukutukhttp/Network.kt
package com.robinyonathan.blukutukhttp

import android.app.Activity
import android.content.Context
import android.net.ConnectivityManager

@Suppress("UNNECESSARY_SAFE_CALL")
internal object Network {
    fun isNetworkAvailable(activity: Activity?): Boolean {
        var connected = false

        activity?.let {
            val connectivityManager = activity.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            connectivityManager?.let {
                val activeNetworkInfo = connectivityManager.activeNetworkInfo
                activeNetworkInfo?.let {
                    connected = activeNetworkInfo.isConnected
                }
            }
        }

        return connected
    }
}
